packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -106,6 +108,11 @@ Iterable<String> _apkFilesFor(AndroidBuildInfo androidBuildInfo) {
   return <String>['app$flavorString-$buildType.apk'];
 }
 
+// The maximum number of Gradle retries.
+// When a build fails due to a network failure, this is the maximum number
+// of times that the build is re-attempted.
+const int kMaxGradleRetries = 10;
```

Comment:
This is effectively a timeout, which I think we should take the opportunity to remove. The way this could work is by letting the binary exponential backoff increase the backoff up to some maximum point (like 10s), and probably some time before then, use the tool's usual message about an operation taking longer than expected. The long operation message could be accompanied by the error text from gradle.

In practice if all these retries fail, CI will hit the job timeout, or the user will ctrl-c the process, which is how we handle other situations like this.